catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js

```diff
@@ -14,210 +14,222 @@
-        },
-        addDisabledConfiguration: function(configuration) {
-            if(this.get("disabledConfigurations") && !this.get("disabledConfigurations").contains(configuration)) {
-                this.get("disabledConfigurations").add(configuration);
-            }
-        },
-        removeConfiguration: function(configuration) {
-            if(this.get("disabledConfigurations").contains(configuration)) {
-                this.stopListening(configuration);
-                this.get("disabledConfigurations").remove(configuration);
-            } else if (configuration === this.get("currentConfiguration")) {
-                this.stopListening(configuration);
-                this.set("currentConfiguration", undefined);
-            }
-        },
-        setCurrentConfiguration: function(configuration) {
-            // check to see if the source already has a 'currentConfiguration'
-            // if it does, make the currentConfiguration disabled and move it to the disabledConfiguration
-            // list
-            var currentConfig = this.get('currentConfiguration');
-            if (currentConfig) {
-                currentConfig.makeDisableCall();
-                this.addDisabledConfiguration(currentConfig);
-            }
-            this.set({currentConfiguration: configuration});
+        Source.Model = Backbone.Model.extend({
+            configUrl: "/jolokia/exec/org.codice.ddf.ui.admin.api.ConfigurationAdmin:service=ui",
+            idAttribute: 'name',
+            initialize: function () {
+                this.set('currentConfiguration', undefined);
+                this.set('disabledConfigurations', new Source.ConfigurationList());
+            },
+            addDisabledConfiguration: function (configuration) {
+                if (this.get("disabledConfigurations") && !this.get("disabledConfigurations").contains(configuration)) {
+                    this.get("disabledConfigurations").add(configuration);
+                }
+            },
+            removeConfiguration: function (configuration) {
+                if (this.get("disabledConfigurations").contains(configuration)) {
+                    this.stopListening(configuration);
+                    this.get("disabledConfigurations").remove(configuration);
+                } else if (configuration === this.get("currentConfiguration")) {
+                    this.stopListening(configuration);
+                    this.set("currentConfiguration", undefined);
+                }
+            },
+            setCurrentConfiguration: function (configuration) {
+                // check to see if the source already has a 'currentConfiguration'
+                // if it does, make the currentConfiguration disabled and move it to the disabledConfiguration
+                // list
+                var currentConfig = this.get('currentConfiguration');
+                if (currentConfig) {
+                    currentConfig.makeDisableCall();
```

Comment:
This isn't a synchronous call. Do you need to be waiting before you add it as a disabled configuration? On a slow connection, it might not actually be disabled.